shared/AppInsightsCore/src/JavaScriptSDK/ChannelController.ts

```diff
@@ -1,135 +1,177 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-"use strict";
+// // Licensed under the MIT License.
 
-import dynamicProto from "@microsoft/dynamicproto-js";
-import { IAppInsightsCore } from "../JavaScriptSDK.Interfaces/IAppInsightsCore"
-import { IConfiguration } from "../JavaScriptSDK.Interfaces/IConfiguration";
+import { IAppInsightsCore } from "../JavaScriptSDK.Interfaces/IAppInsightsCore";
 import { IChannelControls } from "../JavaScriptSDK.Interfaces/IChannelControls";
-import { IPlugin, ITelemetryPlugin  } from "../JavaScriptSDK.Interfaces/ITelemetryPlugin";
-import { ITelemetryPluginChain } from "../JavaScriptSDK.Interfaces/ITelemetryPluginChain";
-import { ITelemetryItem } from "../JavaScriptSDK.Interfaces/ITelemetryItem";
+import { IConfiguration } from "../JavaScriptSDK.Interfaces/IConfiguration";
 import { IProcessTelemetryContext } from "../JavaScriptSDK.Interfaces/IProcessTelemetryContext";
-import { _InternalLogMessage } from "./DiagnosticLogger";
-import { BaseTelemetryPlugin } from "./BaseTelemetryPlugin";
-import { ProcessTelemetryContext } from "./ProcessTelemetryContext";
+import { ITelemetryItem } from "../JavaScriptSDK.Interfaces/ITelemetryItem";
+import { IPlugin } from "../JavaScriptSDK.Interfaces/ITelemetryPlugin";
+import { ITelemetryPluginChain } from "../JavaScriptSDK.Interfaces/ITelemetryPluginChain";
+import { arrForEach, isArray, objFreeze, throwError } from "./HelperFuncs";
+import { createProcessTelemetryContext, createTelemetryProxyChain } from "./ProcessTelemetryContext";
 import { initializePlugins } from "./TelemetryHelpers";
-import { arrForEach, objDefineAccessors, throwError } from "./HelperFuncs";
 
-const ChannelControllerPriority = 500;
-const ChannelValidationMessage = "Channel has invalid priority";
+export const ChannelControllerPriority = 500;
+const ChannelValidationMessage = "Channel has invalid priority - ";
+
+export interface _IInternalChannels {
+    queue: IChannelControls[];
+    chain: ITelemetryPluginChain;
+}
 
-export class ChannelController extends BaseTelemetryPlugin {
```

Comment:
Completely removing the class as it's internal only, so no need to keep it as it should never be exposed.